WordPressKit/HTTPRequestBuilder.swift

```diff
@@ -0,0 +1,140 @@
+import Foundation
+
+final class HTTPRequestBuilder {
+    enum Method: String {
+        case get = "GET"
+        case post = "POST"
+        case put = "PUT"
+        case patch = "PATCH"
+        case delete = "DELETE"
+
+        var allowsHTTPBody: Bool {
+            self == .post || self == .put || self == .patch
+        }
+    }
+    private var urlComponents: URLComponents
+    private var method: Method = .get
+    private var path: String
+    private var headers: [String: String] = [:]
+    private var bodyBuilder: ((inout URLRequest) throws -> Void)?
+
+    init(url: URL) {
+        assert(url.scheme == "http" || url.scheme == "https")
+        assert(url.host != nil)
+
+        urlComponents = URLComponents(url: url, resolvingAgainstBaseURL: true)!
+        path = urlComponents.path
+    }
+
+    func set(method: Method) -> Self {
+        self.method = method
+        return self
+    }
+
+    func set(path: String) -> Self {
+        assert(!path.contains("?") && !path.contains("#"), "Path should not have query or fragment: \(path)")
+
+        self.path = path
+        return self
+    }
+
+    func set(value: String?, forHeader header: String) -> Self {
+        headers[header] = value
+        return self
+    }
+
+    func query(name: String, value: String?, override: Bool = false) -> Self {
+        append(query: [URLQueryItem(name: name, value: value)], override: override)
+    }
+
+    func append(query: [URLQueryItem], override: Bool) -> Self {
+        urlComponents.queryItems = (urlComponents.queryItems ?? []) + query
+        return self
+    }
+
+    func body(form: [String: String]) -> Self {
+        headers["Content-Type"] = "application/x-www-form-urlencoded; charset=utf-8"
+        bodyBuilder = { req in
+            var url = URLComponents(string: "https://wordpress.com")!
```

Comment:
Should this be 

```suggestion
            var url = self.urlComponents
```

I'm surprised `https://wordpress.com/` is hardcoded here but maybe I'm missing something about the intended use? E.g., only requests to the WordPress.com API can send URL form encoded requests?

Reply:
This function needs a URL encoded string. I'm just using a temporary `URLComponents` to construct a query which is url encoded.